third_party/move/move-vm/runtime/src/loader/modules.rs

```diff
@@ -363,8 +365,14 @@ impl Module {
             });
         }
 
+        let mut hasher = blake3::Hasher::new();
```

Comment:
this can be a helper and we can share it with struct name index map? Also, right now this means we hash many times: as when we create a module, we intern all struct names so you are re-computing hash #structs times? We can compute hash for module first, and pass it to the index map?